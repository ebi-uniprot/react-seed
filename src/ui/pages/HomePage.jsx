
import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

import DefaultPageLayout from '../layout/DefaultPageLayout';

const HomePageContent = () => (
  <Fragment>
    <h3>Welcome!</h3>
    <Link to="/sample-page">Go to Sample Page</Link>
  </Fragment>
);

const HomePage = () => (
  <DefaultPageLayout
    title="Home Page"
    content={<HomePageContent />}
  />
);

export default HomePage;
