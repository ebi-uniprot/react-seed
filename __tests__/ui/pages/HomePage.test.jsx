
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import HomePage from '../../../src/ui/pages/HomePage';

describe('HomePage component', () => {
  test('should render', () => {
    const component = renderer
      .create(<MemoryRouter><HomePage /></MemoryRouter>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
