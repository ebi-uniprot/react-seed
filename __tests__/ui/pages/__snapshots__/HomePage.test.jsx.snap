// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render 1`] = `
<div
  className="default-page-layout"
>
  You are viewing 
  <b>
    Home Page
  </b>
  <br />
  <div
    className="default-page-conent"
  >
    <h3>
      Welcome!
    </h3>
    <a
      href="/sample-page"
      onClick={[Function]}
    >
      Go to Sample Page
    </a>
  </div>
</div>
`;
